docs/master/api-reference/directives.md

```diff
@@ -486,13 +486,17 @@ type Query {
 
 ```php
 use Illuminate\Database\Eloquent\Builder;
+use GraphQL\Type\Definition\ResolveInfo;
+use Nuwave\Lighthouse\Support\Contracts\GraphQLContext;
 
 class MyClass
 {
-    public function minimumHighscore(Builder $builder, int $minimumHighscore): Builder
+    public function minimumHighscore(Builder $builder, ?int $minimumHighscore, array $args, GraphQLContext $context, ResolveInfo $resolveInfo): Builder
     {
-        return $builder->whereHas('game', static function (Builder $builder) use ($minimumHighscore): void {
-            $builder->where('score', '>', $minimumHighscore);
+        return $builder->when($minimumHighscore !== null, static function(Builder $builder) use ($minimumHighscore): void {
```

Comment:
Why complicate this by making `minimumHighscore` nullable?

Reply:
@spawnia For that specific example where the user is passing `value: 1000` in schema there is no need for it being nullable you are right, I took it a little more general so if the user doesn't pass a value it wouldn't make:

```php
$builder->whereHas('game', static function (Builder $builder) use ($minimumHighscore): void {
    $builder->where('score', '>', $minimumHighscore);
});
```
to get to be :

```php
$builder->whereHas('game', static function (Builder $builder) use ($minimumHighscore): void {
    $builder->where('score', '>', null);
});
```
Just wanted to make it a little more complete no other reasons, we could revert it or add another example to it for instance:

```graphql
type Query {
    users(
        minimumHighscore: Int @builder(method: "App\MyClass@minimumHighscore")
    ): [User!]! @all
    highrankedUsers: [User!]! @all @builder(method: "App\MyClass@minimumHighscore")
}
```

so it would make sense.

Reply:
I figured out the value could be `null` even in the current example - when the client explicitly passed `minimumHighscore: null`. I modified the example.